tracer/src/Datadog.Trace/Ci/GitCommandHelper.cs

```diff
@@ -0,0 +1,191 @@
+    private static readonly Regex DiffHeaderRegex = new Regex(@"^diff --git a/(?<file>.+) b/(?<file2>.+)$");
+    private static readonly Regex LineChangeRegex = new Regex(@"^@@ -\d+(,\d+)? \+(?<start>\d+)(,(?<count>\d+))? @@");
+    private static char[] lineSeparators = { '\n', '\r' };
+
+    public static async Task<ProcessHelpers.CommandOutput?> RunGitCommandAsync(string? workingDirectory, string arguments, MetricTags.CIVisibilityCommands ciVisibilityCommand, string? input = null)
+    {
+        TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommand(ciVisibilityCommand);
+        try
+        {
+            var sw = System.Diagnostics.Stopwatch.StartNew();
+            var gitOutput = await ProcessHelpers.RunCommandAsync(
+                                new ProcessHelpers.Command(
+                                    "git",
+                                    arguments,
+                                    workingDirectory,
+                                    outputEncoding: Encoding.Default,
+                                    errorEncoding: Encoding.Default,
+                                    inputEncoding: Encoding.Default,
+                                    useWhereIsIfFileNotFound: true),
+                                input).ConfigureAwait(false);
+            TelemetryFactory.Metrics.RecordDistributionCIVisibilityGitCommandMs(ciVisibilityCommand, sw.Elapsed.TotalMilliseconds);
+            if (gitOutput is null)
+            {
+                TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommandErrors(ciVisibilityCommand, MetricTags.CIVisibilityExitCodes.Unknown);
+                Log.Warning("GitCommand: 'git {Arguments}' command is null", arguments);
+            }
+            else if (gitOutput.ExitCode != 0)
+            {
+                TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommandErrors(MetricTags.CIVisibilityCommands.GetRepository, TelemetryHelper.GetTelemetryExitCodeFromExitCode(gitOutput.ExitCode));
+            }
+
+            if (Log.IsEnabled(Vendors.Serilog.Events.LogEventLevel.Debug))
+            {
+                Log.Debug("Git command : {Command}", $"git {arguments}");
+                Log.Debug("  exit code : {Output}", gitOutput?.ExitCode);
+                Log.Debug("     output : {Output}", gitOutput?.Output ?? "<NULL>");
+                if (gitOutput is not null && gitOutput.Error is { Length: > 0 } err)
+                {
+                    Log.Debug("     error  : {Error}", err);
+                }
+            }
+
+            return gitOutput;
+        }
+        catch (System.ComponentModel.Win32Exception ex)
+        {
+            Log.Warning(ex, "GitCommand: 'git {Arguments}' threw Win32Exception - git is likely not available", arguments);
+            TelemetryFactory.Metrics.RecordCountCIVisibilityGitCommandErrors(ciVisibilityCommand, MetricTags.CIVisibilityExitCodes.Missing);
+            return null;
+        }
+    }
```

Comment:
This need to change, now there's no more async version of the process helper so this will simplify, also beware there's a new timeout parameter in the command ctor that we are using, so I recommend you to port this code again from the original location.